docs/streams/developer-guide.html

```diff
@@ -267,21 +295,532 @@
+
+             // Java 7 example
+             KStream&lt;String, Long&gt; onlyPositives = stream.filterNot(
+              new Predicate&lt;String, Long&gt;() {
+                @Override
+                public boolean test(String key, Long value) {
+                    return value <= 0;
+                }
+             });
+         </pre>
+	      </td>
+	 </tr>
+	 <tr>
+	   <td><b>FlatMap</b>: <code>KStream->KStream </code></td>
+         <td>
+             Takes one record and produces zero, one, or more records. You can modify the record keys and values,
+             including their types.<p>
+
+             Marks the stream for data re-partitioning: Applying a grouping or a join after <code>flatMap</code> will result in
+             re-partitioning of the records. If possible use <code>flatMapValues</code> instead, which will not cause data re-partitioning.
+
+             <pre class="brush: java;">
+             KStream&lt;Long, String> stream = ...;
+             KStream&lt;String, Integer&gt; transformed = stream.flatMap(
+                 // Here, we generate two output records for each input record.
+                 // We also change the key and value types.
+                 // Example: (345L, "Hello") -> ("HELLO", 1000), ("hello", 9000)
+                 (key, value) -> {
+                     List&lt;KeyValue&lt;String, Integer&gt;&gt; result = new LinkedList&lt;&gt;();
+                     result.add(KeyValue.pair(value.toUpperCase(), 1000));
+                     result.add(KeyValue.pair(value.toLowerCase(), 9000));
+                     return result;
+                 }
+             );
+             // Java 7 example: cf. `map` for how to create `KeyValueMapper` instances
+         </pre>
+         </td>
+     </tr>
+       <tr>
+           <td><b>FlatMap (values only)</b>: <code>KStream->KStream </code></td>
+           <td>
+               Takes one record and produces zero, one, or more records, while retaining the key of the original record.
+               You can modify the record values and the value type.
+
+               <code>flatMapValues</code> is preferable to <code>flatMap</code> because it will not cause data re-partitioning. However,
+               it does not allow you to modify the key or key type like <code>flatMap</code> does.
+
+               <pre class="brush: java;">
+               // Split a sentence into words.
+               KStream&lt;byte[], String&gt; sentences = ...;
+               KStream&lt;byte[], String&gt; words = sentences.flatMapValues(value -> Arrays.asList(value.split("\\s+")));
+
+               // Java 7 example: cf. `mapValues` for how to create `ValueMapper` instances
+               </pre>
+           </td>
+       </tr>
+       <tr>
+           <td><b>Foreach</b>: <code>KStream->void or KTable->void </code></td>
+           <td>
+               Terminal operation. Performs a stateless action on each record.<p>
```

Comment:
"Terminal operation." should be in bold font.  Same for other instances of "Terminal operation."